src/accelerate/test_utils/scripts/test_script.py

```diff
@@ -280,7 +287,7 @@ def central_dl_preparation_check():
 def mock_training(length, batch_size, generator):
     set_seed(42)
     generator.manual_seed(42)
-    train_set = RegressionDataset(length=length)
+    train_set = RegressionDataset(length=length, seed=42)
```

Comment:
See https://github.com/huggingface/accelerate/pull/1385#discussion_r1186414407

Each thread needs to make a `Generator` with a common seed, rather than using the global random state.